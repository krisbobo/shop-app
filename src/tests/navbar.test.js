import { render } from '@testing-library/react';
import { useSelector } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Navbar from '../components/navbar';

jest.mock('react-redux');

describe('Test Navbar Component', () => {
  test('Should render navbar', () => {
    useSelector.mockReturnValue({
      categories: ['electronics', 'jewelery', "men's clothing", "women's clothing"],
      isActive: false,
      isLoading: false,
      categoryTitle: 'All Products',
    });
    const tree = render(
      <BrowserRouter>
        <Navbar />
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
