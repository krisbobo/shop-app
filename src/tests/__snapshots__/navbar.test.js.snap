// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Navbar Component Should render navbar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="navbar__Header-sc-z3rhhf-0 eENQcZ"
      >
        <nav
          class="navbar__Container-sc-z3rhhf-1 CjPTj"
        >
          <div
            class="navbar__BurgerInnerContainer-sc-z3rhhf-7 urpkD"
          >
            <h1
              class="navbar__Logo-sc-z3rhhf-2 hLcLWM"
            >
              <a
                aria-current="page"
                class="active"
                href="/"
              >
                Shop
              </a>
            </h1>
            <button
              class="navbar__BurgerButton-sc-z3rhhf-6 guZOZs"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
                />
              </svg>
            </button>
          </div>
          <div
            class="navbar__BurgerNavContainer-sc-z3rhhf-3 iVkhUn"
          >
            <button
              class="navbar__CloseButton-sc-z3rhhf-8 fuUwpC"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                />
              </svg>
            </button>
            <ul
              class="navbar__NavLinks-sc-z3rhhf-4 dwnUBG"
            >
              <li
                class="navbar__Li-sc-z3rhhf-5 dhZBbA"
              >
                electronics
              </li>
              <li
                class="navbar__Li-sc-z3rhhf-5 dhZBbA"
              >
                jewelery
              </li>
              <li
                class="navbar__Li-sc-z3rhhf-5 dhZBbA"
              >
                men's clothing
              </li>
              <li
                class="navbar__Li-sc-z3rhhf-5 dhZBbA"
              >
                women's clothing
              </li>
            </ul>
          </div>
        </nav>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="navbar__Header-sc-z3rhhf-0 eENQcZ"
    >
      <nav
        class="navbar__Container-sc-z3rhhf-1 CjPTj"
      >
        <div
          class="navbar__BurgerInnerContainer-sc-z3rhhf-7 urpkD"
        >
          <h1
            class="navbar__Logo-sc-z3rhhf-2 hLcLWM"
          >
            <a
              aria-current="page"
              class="active"
              href="/"
            >
              Shop
            </a>
          </h1>
          <button
            class="navbar__BurgerButton-sc-z3rhhf-6 guZOZs"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M32 96v64h448V96H32zm0 128v64h448v-64H32zm0 128v64h448v-64H32z"
              />
            </svg>
          </button>
        </div>
        <div
          class="navbar__BurgerNavContainer-sc-z3rhhf-3 iVkhUn"
        >
          <button
            class="navbar__CloseButton-sc-z3rhhf-8 fuUwpC"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </button>
          <ul
            class="navbar__NavLinks-sc-z3rhhf-4 dwnUBG"
          >
            <li
              class="navbar__Li-sc-z3rhhf-5 dhZBbA"
            >
              electronics
            </li>
            <li
              class="navbar__Li-sc-z3rhhf-5 dhZBbA"
            >
              jewelery
            </li>
            <li
              class="navbar__Li-sc-z3rhhf-5 dhZBbA"
            >
              men's clothing
            </li>
            <li
              class="navbar__Li-sc-z3rhhf-5 dhZBbA"
            >
              women's clothing
            </li>
          </ul>
        </div>
      </nav>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
